### PR TITLE
Support gfx908

### PR DESCRIPTION
**Pull request by Serge45**
## Brief ##
This PR attempts to make  `hipBLASLt` support gfx908.

## Tasks ##
 - [x] Added basic YAMLs that make test suites pass on gfx908 platform([test result](https://github.com/ROCmSoftwarePlatform/hipBLASLt/files/12651491/test_result.txt))
 - [x] Use regex instead of hand-written wild-card matching to specify the platforms that tests should run on.

### CODE REVIEW

File: tensilelite/Tensile/Ops/SoftmaxGenerator.py [hunk 1]
```diff
@@ -605,6 +605,16 @@ def kernel_rodata(name: str):
 .amdhsa_next_free_vgpr .amdgcn.next_free_vgpr
 .amdhsa_next_free_sgpr .amdgcn.next_free_sgpr
 .end_amdhsa_kernel
+''' if gfx_arch >= (9, 0, 10) else\
```
Comment (TonyYHsieh) [inline tensilelite/Tensile/Ops/SoftmaxGenerator.py:608]: Nai21 = gfx1030, Navi31 = gfx1100

Comment (Serge45) [reply tensilelite/Tensile/Ops/SoftmaxGenerator.py:608]: Almost forgot about these archs, will fix soon, thanks.

**Review by KKyang — APPROVED** [review]

**Comment by IMbackK** [comment]
Could we get this merged?
It seams to work fine and it seams pointless to let sutch a usefull feature languish in a pr endlessly

**Comment by Serge45** [comment]
Rebased on develop branch with minor tweaks, local tests are all passed. Here's the testing log: [test_result_20240715.txt](https://github.com/user-attachments/files/16229942/test_result_20240715.txt)

**Comment by IMbackK** [comment]
I just did some basic tests with the rebase here and it seams fine on my end.

**Review by KKyang — APPROVED** [review]

**Review by TonyYHsieh — APPROVED** [review]
LGTM

**Comment by LunNova** [comment]
Is anything special needed to use this support? I'm trying to build for gfx908 after this PR (on commit 3c44762350b2386f3a0f448cfb732f3eee380f2b) and getting

```
Creating code object for arch gfx908
/build/source/build/library/build_tmp/ops/A_S_H_256_4_gfx908.s:109:1: error: operands are not valid for this GPU or mode
v_add_u32 v[vgprOffset+1], vcc, v[vgprOffset+0], s[sgprTmp]
^
/build/source/build/library/build_tmp/ops/A_S_H_256_4_gfx908.s:110:1: error: operands are not valid for this GPU or mode
v_add_u32 v[vgprOffset+2], vcc, v[vgprOffset+1], s[sgprTmp]
^
/build/source/build/library/build_tmp/ops/A_S_H_256_4_gfx908.s:111:1: error: operands are not valid for this GPU or mode
v_add_u32 v[vgprOffset+3], vcc, v[vgprOffset+2], s[sgprTmp]
^
/build/source/build/library/build_tmp/ops/A_S_H_256_4_gfx908.s:121:90: error: 'null' operand is not supported on this GPU
buffer_load_dwordx4 v[vgprValue+0:vgprValue+0+3], v[vgprOffset+0], s[sgprSrc:sgprSrc+3], null offen offset:0
                                                                                         ^
/build/source/build/library/build_tmp/ops/A_S_H_256_4_gfx908.s:122:90: error: 'null' operand is not supported on this GPU
buffer_load_dwordx4 v[vgprValue+4:vgprValue+4+3], v[vgprOffset+1], s[sgprSrc:sgprSrc+3], null offen offset:0
                                                                                         ^
/build/source/build/library/build_tmp/ops/A_S_H_256_4_gfx908.s:123:90: error: 'null' operand is not supported on this GPU
buffer_load_dwordx4 v[vgprValue+8:vgprValue+8+3], v[vgprOffset+2], s[sgprSrc:sgprSrc+3], null offen offset:0
                                                                                         ^
/build/source/build/library/build_tmp/ops/A_S_H_256_4_gfx908.s:124:92: error: 'null' operand is not supported on this GPU
buffer_load_dwordx4 v[vgprValue+12:vgprValue+12+3], v[vgprOffset+3], s[sgprSrc:sgprSrc+3], null offen offset:0
                                                                                           ^
/build/source/build/library/build_tmp/ops/A_S_H_256_4_gfx908.s:149:1: error: operands are not valid for this GPU or mode
v_add_u32 v[vgprOffset+0], vcc, v[vgprOffset+0], s[sgprTmp]
^
/build/source/build/library/build_tmp/ops/A_S_H_256_4_gfx908.s:150:1: error: operands are not valid for this GPU or mode
v_add_u32 v[vgprOffset+1], vcc, v[vgprOffset+1], s[sgprTmp]
^
```

Full build log: https://gist.github.com/LunNova/692b994279722b1aa506496aa8e0def2

**Pull request merged**